### PR TITLE
Feature/control max concurrent connections

### PR DESCRIPTION
**Pull request by mb-ca**
Adds the ability to limit the total number of concurrent requests sent to the ChannelApe API at one time.  After the configured pool size is used up, requests are placed into a FIFO queue.  As a currently processing request completes, the next request in the queue is started.

### CODE REVIEW

File: test/RequestClientWrapper.spec.ts [hunk 1]
```diff
@@ -579,5 +580,142 @@ Code: 0 Message: You didnt pass any body`;
...
+        });
+      });
+
+      it('When making 2 requests at the same time', (done) => {
+        const TIMEOUT = 100;
+        sandbox
+          .stub(axios, 'get')
+          .callsFake(() => {
+            return new Promise((resolve) => {
+              setTimeout(resolve, 100);
```
Comment (rjdavis3) [inline test/RequestClientWrapper.spec.ts]: I'm not a big fan of the time based unit tests but I understand that this is difficult to unit test. Another option could have been exposing `requestQueue` and `pendingRequests` instead of keeping them private so that this unit test could

File: src/RequestClientWrapper.ts [hunk 2]
```diff
@@ -16,58 +17,76 @@ interface CallDetails {
 
 export default class RequestClientWrapper {
 
+  private readonly maximumConcurrentConnections: number;
   private readonly requestLogger: RequestLogger;
+  private requestQueue: any;
```
Comment (rjdavis3) [inline src/RequestClientWrapper.ts]: Please add a type to this:
private requestQueue: RequestConfig[];

File: test/RequestClientWrapper.spec.ts (hunk 1 above)
Comment (rjdavis3) [reply test/RequestClientWrapper.spec.ts]: Please spend a little more time here to see if you can remove the time based aspect of these tests since they could break depending on what machine they are running on. If not we can go up with what you have but I won't be surprised if we need to refactor this later.

Comment (ryankazokas) [reply test/RequestClientWrapper.spec.ts]: Agreed @rjdavis3 @mb-ca . This is actually failing on my machine and the test is failing silently since done is never called properly. Only found it through logging:

`{ Error:  /v1/orders/1
  Status: 0
  Response Body:
  expected 162 to be below 125
    at requestPromise.then.catch (c:\Development\channelape-typescript-sdk\src\RequestClientWrapper.ts:158:34)
    at <anonymous>
  apiErrors: [],
  responseStatusCode: -1,
  responseStatusMessage: 'There was an error with the API' }`

**Comment by ryankazokas** [comment]
@rjdavis3 this ready for review/merge. Also per our discussion i've removed stryker mutation tests in favor of forcing a minimum line/branch coverage. In the posttest command you'll see is where i added the thresholds for nyc to check coverage. Looks like we are already above that for everything:

=============================== Coverage summary ===============================
Statements   : 97.44% ( 1749/1795 )
Branches     : 90.98% ( 353/388 )
Functions    : 95.24% ( 240/252 )
Lines        : 97.41% ( 1727/1773 )

To test this you can run npm run cover locally with only a single unit test(.only). You'll see the command will return a -1 result and fail so this should fail the travis build as well.
